### PR TITLE
[Alerting]Fix pod restart not fire alerting after server restarted

### PR DESCRIPTION
**Pull request by aiwantaozi**
Problem:
pod restart count cache was stored in cache when alert create, when server restart the cache disappeared, so couldn't get pre pod restart count
Solution:
store current pod restart count in the cache when check pod restart

Issue:
https://github.com/rancher/rancher/issues/20339

### CODE REVIEW

File: pkg/controllers/user/alert/watcher/pod.go [hunk 1]
```diff
@@ -156,86 +156,82 @@ func (w *PodWatcher) watchRule() error {
...
 }
 
 func (w *PodWatcher) getRestartTimeFromTrack(alert *v3.ProjectAlertRule, curCount int32) int32 {
 	name := alert.Name
 	namespace := alert.Namespace
+	now := time.Now()
+	currentRestartTrack := restartTrack{Count: curCount, Time: now}
+	currentRestartTrackArr := []restartTrack{currentRestartTrack}
 
 	obj, ok := w.podRestartTrack.Load(namespace + ":" + name)
```
Comment (orangedeng) [inline pkg/controllers/user/alert/watcher/pod.go]: Should we use `LoadOrStore`?

File: pkg/controllers/user/alert/watcher/pod.go [hunk 2]
```diff
@@ -156,86 +156,82 @@ func (w *PodWatcher) watchRule() error {
...
 	if !ok {
+		w.podRestartTrack.Store(namespace+":"+name, currentRestartTrackArr)
```
Comment (orangedeng) [inline pkg/controllers/user/alert/watcher/pod.go]: Do we really need to store the whole track in the map? I think you just need the latest one. So maybe we just need to store that.

**Comment by orangedeng** [comment]
LGTM

File: pkg/controllers/user/alert/watcher/pod.go [hunk 3]
```diff
@@ -156,86 +155,76 @@ func (w *PodWatcher) watchRule() error {
...
-			tracks = tracks[i:]
-			tracks = append(tracks, restartTrack{Count: curCount, Time: now})
-			w.podRestartTrack.Store(namespace+":"+name, tracks)
-			return track.Count
+	currentRestartTrack := restartTrack{Count: curCount, Time: now}
+	currentRestartTrackArr := []restartTrack{currentRestartTrack}
+
+	obj, loaded := w.podRestartTrack.LoadOrStore(namespace+":"+name, currentRestartTrackArr)
+	if loaded {
+		tracks := obj.([]restartTrack)
```
Comment (alena1108) [inline pkg/controllers/user/alert/watcher/pod.go]: Looks like formatting is off. It is more than 4 spaces

Comment (aiwantaozi) [reply pkg/controllers/user/alert/watcher/pod.go]: Hi Alena, it's 4 spaces, may the web preview looks like more than 4 spaces

**Comment by cjellick** [comment]
will let @alena1108 finish the review

**Comment by alena1108** [comment]
LGTM

**Pull request merged**